llama.cpp

```diff
@@ -9841,22 +9860,35 @@ void llama_backend_init(bool numa) {
 #ifdef GGML_USE_MPI
     ggml_mpi_backend_init();
 #endif
+
+#ifdef GGML_USE_KOMPUTE
+    if (!ggml_vk_has_device()) {
+        ggml_vk_init_device(0, "gpu");
+    }
+#endif
```

Comment:
Repeating this here: since it looks like the fix was reverted, I will raise the same issue again. It is not good to require that applications call backend-specific functions before the backend can be used. I don't see why this can't be done automatically on the first call to the ggml-backend APIs.